### PR TITLE
Kconfig: expose IEEE 802.15.4 Security to Kconfig

### PR DESCRIPTION
**Pull request by fabian18**
### Contribution description
Makes IEEE 802.15.4 security module configurable with Kconfig.

### Testing procedure
`USEMODULE="ieee802154 ieee802154_security" BOARD=nucleo-f767zi make -C examples/gnrc_networking menuconfig`

Screenshot_20210531_101355

Screenshot_20210531_101435

Screenshot_20210531_101849

### CODE REVIEW

File: sys/include/net/ieee802154_security.h [hunk 1]
```diff
@@ -101,14 +101,14 @@ struct ieee802154_sec_dev {
...
-#if !defined(IEEE802154_SEC_DEFAULT_KEY) || defined(DOXYGEN)
+#if !defined(CONFIG_IEEE802154_SEC_DEFAULT_KEY) || defined(DOXYGEN)
 /**
  * @brief   AES key that is used in the test vectors from the specification
  *
  * @note    Predefine it yourself,
  *          if you want another key to be set up on initialization
  */
-#define IEEE802154_SEC_DEFAULT_KEY              { 0xc0, 0xc1, 0xc2, 0xc3,   \
+#define CONFIG_IEEE802154_SEC_DEFAULT_KEY       { 0xc0, 0xc1, 0xc2, 0xc3,   \
```
Comment (leandrolanzieri) [inline sys/include/net/ieee802154_security.h:111]: Could we have the same default value in both the macro and Kconfig version?

Comment (fabian18) [reply sys/include/net/ieee802154_security.h:111]: Sure. Do you know if Kconfig has a byte array type? I did not find that type when I searched on google so I made it a string.
Or should I make `CONFIG_IEEE802154_SEC_DEFAULT_KEY` macro a string as well?

Comment (leandrolanzieri) [reply sys/include/net/ieee802154_security.h:111]: > Sure. Do you know if Kconfig has a byte array type? I did not find that type when I searched on google so I made it a string.

No, it does not have such a type.

> Or should I make `CONFIG_IEEE802154_SEC_DEFAULT_KEY` macro a string as well?

I see no reason not to make it a string as well

**Review by leandrolanzieri — APPROVED** [review]
ACK

**Comment by fabian18** [comment]
Thank You

**Pull request merged**